ui/ozone/platform/x11/x11_display_manager.cc

```diff
@@ -0,0 +1,335 @@
+// Copyright (c) 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ui/ozone/platform/x11/x11_display_manager.h"
+
+#include "base/command_line.h"
+#include "base/logging.h"
+#include "base/memory/protected_memory_cfi.h"
+#include "base/strings/stringprintf.h"
+#include "ui/aura/window_event_dispatcher.h"
+#include "ui/base/x/x11_util.h"
+#include "ui/display/display.h"
+#include "ui/display/util/display_util.h"
+#include "ui/display/util/x11/edid_parser_x11.h"
+#include "ui/events/platform/platform_event_source.h"
+#include "ui/gfx/icc_profile.h"
+#include "ui/gfx/switches.h"
+#include "ui/gfx/x/x11.h"
+#include "ui/gfx/x/x11_atom_cache.h"
+#include "ui/gfx/x/x11_types.h"
+
+#include <dlfcn.h>
+
+namespace ui {
+
+namespace {
+
+constexpr int default_refresh = 60;
+
+// static
+gfx::ICCProfile GetICCProfileForMonitor(int monitor) {
```

Comment:
Can we share this and some code below with the stock x11? Just copying the code doesn't look nice. Otherwise, you can do it while upstreaming the patch.  up to you